api/test/integration/geo-coding/geo-coding.service.spec.ts

```diff
@@ -53,6 +57,79 @@ describe('GeoCoding Service (Integration Testing)', () => {
   });
 
   describe('Unknown Location type tests', () => {
+    test('When a unknown location type without country is sent, proper error message must be received', async () => {
+      jest
+        .spyOn(adminRegionService, 'getAdminAndGeoRegionIdByCountryIsoAlpha2')
+        .mockResolvedValue({
+          id: 'ddb17f37-4e6a-4494-95c8-26ed830317df',
+          geoRegionId: 'ddb17f37-4e6a-4494-95c8-26ed830317df',
+        });
+      jest
+        .spyOn(sourcingLocationService, 'save')
+        .mockResolvedValue([] as unknown as SourcingLocation[]);
+      jest
+        .spyOn(unknownLocationService, 'geoCodeByCountry')
+        .mockResolvedValue(geocodeResponses[4] as GeocodeResponseData);
+      const sourcingData = {
+        locationCountryInput: null,
+      } as unknown as SourcingData;
+
+      expect.assertions(1);
+      try {
+        await geoCodingService.geoCodeUnknownLocationType(sourcingData);
+      } catch ({ message }) {
+        expect(message).toEqual(
+          'A country where material is received needs to be provided for Unknown Location Types',
+        );
+      }
+    });
+
+    test('When a unknown location type with coordinates or address is sent, proper error message must be received', async () => {
+      jest
+        .spyOn(adminRegionService, 'getAdminAndGeoRegionIdByCountryIsoAlpha2')
+        .mockResolvedValue({
+          id: 'ddb17f37-4e6a-4494-95c8-26ed830317df',
+          geoRegionId: 'ddb17f37-4e6a-4494-95c8-26ed830317df',
+        });
+      jest
+        .spyOn(sourcingLocationService, 'save')
+        .mockResolvedValue([] as unknown as SourcingLocation[]);
+      jest
+        .spyOn(unknownLocationService, 'geoCodeByCountry')
+        .mockResolvedValue(geocodeResponses[4] as GeocodeResponseData);
+      const sourcingDataWithCoordinates = {
+        locationCountryInput: 'Asgard',
+        locationLongitude: 78.96288,
+        locationLatitude: 20.593684,
+      } as unknown as SourcingData;
+
+      const sourcingDataWithAddress = {
+        locationCountryInput: 'Asgard',
+        locationAddress: 'Valhalla, 3',
+      } as unknown as SourcingData;
+
+      expect.assertions(1);
```

Comment:
Is this test actually working OK? 
So the goal of this expect.assertions(number) is to actually expect a determined number of assertions (captain obvious here).

So in example: if you write a test with no ```expect(something).toSomething()``` The test will actually 'pass'.

This being said, if we expect a error to be thrown (that is inside a catch block scope), and there is no error thrown, it will never reach the expect statement and effectivelly will mean that 'there is no expect' in this test, therefore passing the test and giving us a false positive. ```expect.assertions(number``` is our guard here, if the test reach no expect, but if we state that we actually expect some expect (lol), the test will fail.

So I see 2 assertions here, but expecting just one. So either one of said expect is not being reached, or there is something wrong

Am I missing something?